src/text.c

```diff
@@ -1,48 +1,309 @@
+EWRAM_DATA static OpenedFile *sCharmapFiles[3] = {NULL};
+EWRAM_DATA static UnkTextStruct2 sUnknown_202AFC0[4] = {0};
+EWRAM_DATA static s32 sUnknown_202B020 = 0;
+EWRAM_DATA static s32 sUnknown_202B024 = 0;
+EWRAM_DATA static s32 sCharHeight[2] = {0};
+EWRAM_DATA static u32 sTextShadowMask = 0; // Some text color info is stored; retrieve via "& 0xF"
+EWRAM_DATA static u8 sDrawTextShadow = 0;
+EWRAM_DATA ALIGNED(4) u16 gUnknown_202B038[4][32][32] = {0};
+
+static void SaveUnkTextStructAndXXX_8006438(const UnkTextStruct2 *a0, bool8 a1, bool8 a2, UnkTextStruct2_sub *a3);
+static void sub_8006554(UnkTextStruct1 *a0, u32 *a1, u32 *a2, u16 *a3, u32 a4, const UnkTextStruct2 *a5, bool8 a6, u32 a7, UnkTextStruct2_sub *a8, u8 a9);
+static void sub_800677C(UnkTextStruct1 *a0, s32 a1, u16 *a2, u8 a3);
+static void sub_80069CC(UnkTextStruct1 *a0, s32 a1, s32 a2, s32 a3, u16 *a4);
+static void sub_8006AC4(UnkTextStruct1 *a0, s32 a1, s32 a2, s32 a3, u16 *a4);
+static void sub_8006B70(UnkTextStruct1 *a0, s32 a1, s32 a2, s32 a3, u16 *a4);
+static void sub_8006C44(UnkTextStruct1 *a0, s32 a1, u16 *a2, u8 a3);
+static void sub_8006E94(UnkTextStruct1 *a0, s32 a1, u32 a2, const UnkTextStruct2_sub2 *a3, u16 *a4);
+static u32 xxx_draw_char(struct UnkTextStruct1 *a0, s32 x, s32 y, u32 a3, u32 color, u32 a5);
+static void nullsub_129(u32 a0, s32 x, s32 y, s32 a3, u32 color);
+static void sub_8007958(UnkTextStruct1 *a0, u32 a1, s32 x, s32 y, s32 a4, u32 color);
+static void sub_8007AA4(struct UnkTextStruct1 *a0, u32 a1, s32 x, s32 y, s32 a4, u32 color);
+static void sub_8007BA8(UnkTextStruct1 *a0, u32 a1, s32 x, s32 y, s32 a4, s32 color);
+static void sub_8007D00(UnkTextStruct1 *a0, u32 a1, s32 x, s32 y, s32 a4, s32 color);
+static void sub_8007E64(UnkTextStruct1 *a0, u16 a1[32][32], u32 a2, s32 a3, s32 a4, s32 a5, s32 a6, u32 *a7, u32 a8);
+static void sub_8008030(UnkTextStruct1 *a0, u16 a1[32][32], u32 a2, s32 a3, s32 a4, s32 a5, s32 a6, u32 *a7, u32 a8);
+static void DisplayMonPortrait(UnkTextStruct1 *a0, u16 a1[32][32], s32 a2, const u8 *compressedData, u32 a4);
+static void DisplayMonPortraitFlipped(UnkTextStruct1 *a0, s32 a1, const u8 *compressedData, s32 a3);
+static void sub_80084A4(UnkTextStruct1 *a0, u16 a1[32][32], u32 a2, s32 a3, s32 a4, s32 a5, s32 a6, u32 a8);
+static void sub_8008818(UnkTextStruct1 *a0, s32 a1, s32 a2, s32 a3, s32 a4, s32 a5);
+static bool8 xxx_update_bg_vram(UnkTextStruct1 *a0);
+static void sub_800898C(void);
+static void sub_80089AC(const UnkTextStruct2 *r4, UnkTextStruct2_sub *r5_Str);
+static void PrepareTextbox_8008C6C(UnkTextStruct1 *strArr, u32 strId);
+static void xxx_draw_string(UnkTextStruct1 *strArr, s32 x, s32 y, const u8 *str, u32 windowId, u32 terminatingChr, s32 characterSpacing, s32 lineSpacing);
+static const u8 *HandleTextFormat(UnkTextStruct1 *strArr, const u8 *str, struct UnkDrawStringStruct *sp);
+static void sub_8009388(void);
+static s32 InterpretColorChar(u8 a0);
+
+UNUSED static const char sPksDir0[] = "pksdir0";
```

Comment:
This is mainly handled with the `globaldata.h`. Two of them here almost implies another file split we are missing based on information from @Kermalis.

Reply:
These are one after the other, so not sure any file is missing. Could just be something with how these pksdir things were generated.
Looking at const and ewram data file splitting makes little sense here imo

Reply:
I could put one in `random.c` which is just before `text.c`

Reply:
Sure, that would work